tmt/steps/provision/minute.py

```diff
@@ -0,0 +1,250 @@
+            click.option(
+                '--openstack', metavar='URL', required=True,
+                help='URL of openstack API to use for the provision.'),
+            ] + super().options(how)
+
+    def _get_default_image(self):
+        _, images = run_openstack(self.get('openstack'),
+                'image list -f value -c Name', True)
+        # Use the latest Fedora image
+        fedora_re = re.compile(r'1MT-Fedora-(?P<ver>\d+)')
+        fedora_images = [image for image in images.splitlines() if fedora_re.match(image)]
+        return max(fedora_images, key=lambda x: int(fedora_re.match(x).group('ver')))
+    
+    def default(self, option, default=None):
+        """ Return the default value for the given option """
+        defaults = {
+            'image': self._get_default_image(),
+            'flavor': 'm1.small'
+        }
+        return defaults.get(option, default)
+
+    def show(self):
+        """ Show provision details """
+        super().show(['image', 'flavor'])
+
+    def wake(self, data=None):
+        """ Override options and wake up the guest """
+        super().wake(['image', 'flavor', 'openstack'])
+        if data:
+            self._guest = Guest1minutetip(data, name=self.name, parent=self.step)
+            self._guest.wake()
+
+    def go(self):
+        """ Provision the container """
+        super().go()
+
+        data = dict(user=DEFAULT_USER)
+        for opt in ['image', 'flavor', 'openstack']:
+            val = self.get(opt)
+            if opt != 'openstack':
+                self.info(opt, val, 'green')
+            data[opt] = val
+
+        self._guest = Guest1minutetip(data, name=self.name, parent=self.step)
+        self._guest.start()
+
+    def guest(self):
+        """ Return the provisioned guest """
+        return self._guest
+
+
+class Guest1minutetip(tmt.Guest):
+    """
+    1minutetip instance
+
+    The following keys are expected in the 'data' dictionary:
+        
+        image ...... 1minutetip image name
+        flavor ..... openstack server flavor to use
+        openstack .. URL of openstack API
+    """
+    def load(self, data):
+        super().load(data)
+        self.key = SSH_KEY
+        self.openstack = data.get('openstack')
+        self.image = data.get('image')
+        self.flavor = data.get('flavor')
+        self.username = getpass.getuser()
+        self.instance_name = data.get('instance')
+
+    def save(self):
+        data = super().save()
+        data['openstack'] = self.openstack
+        data['instance'] = self.instance_name
+        data['image'] = self.image
+        data['flavor'] = self.flavor
+        return data
+
+    def _guess_net_id(self):
+        _, networks = run_openstack(self.openstack, 'ip availability list -f json')
+        networks = json.loads(networks)
+        networks = [network for network in networks if
+                re.match(r'provider\_net\_cci', network['Network Name'])]
```

Comment:
Please create a global variable with this regexp, it can be possible we will need to change it in the future. Ideally it should be in config and not hardcoded, but I guess we have no really better option right now